tests/pytest/test_binary_value.py

```diff
@@ -111,25 +111,39 @@ def test_init_unsigned_negative_value():
         pytest.fail("Expected ValueError when assigning negative number to unsigned BinaryValue")
 
 
-def test_init_not_enough_bits(capsys):
-    bin1 = BinaryValue(value=128, n_bits=7, bigEndian=True, binaryRepresentation=BinaryRepresentation.UNSIGNED)
-    assert bin1._str == "0000000"
-    assert bin1.binstr == "0000000"
-    assert bin1.integer == 0
-    captured = capsys.readouterr()
-    assert captured.out == "WARNING: truncating value to match requested number of bits (8 -> 7)\n"
-
-    bin2 = BinaryValue(value="1111110000101100", n_bits=12, bigEndian=False,
-                       binaryRepresentation=BinaryRepresentation.TWOS_COMPLEMENT)
-    assert bin2._str == "110000101100"
-    assert bin2.binstr == "110000101100"
-    assert bin2.integer == -980
-
-    bin3 = BinaryValue(value="1111110000101100", n_bits=11, bigEndian=False,
-                       binaryRepresentation=BinaryRepresentation.SIGNED_MAGNITUDE)
-    assert bin3._str == "10000101100"
-    assert bin3.binstr == "10000101100"
-    assert bin3.integer == -44
+def test_init_not_enough_bits():
+    with pytest.warns(RuntimeWarning, match=r"\d+-bit value requested, truncating value"):
```

Comment:
You should have one of these around _each_ line you expect to warn

Reply:
Done.